src/gui/search/searchtab.cpp

```diff
@@ -78,27 +83,49 @@ SearchTab::SearchTab(SearchWidget *parent)
     m_proxyModel = new SearchSortModel(this);
     m_proxyModel->setDynamicSortFilter(true);
     m_proxyModel->setSourceModel(m_searchListModel);
-    m_resultsBrowser->setModel(m_proxyModel);
+    m_ui->resultsBrowser->setModel(m_proxyModel);
 
     m_searchDelegate = new SearchListDelegate(this);
-    m_resultsBrowser->setItemDelegate(m_searchDelegate);
+    m_ui->resultsBrowser->setItemDelegate(m_searchDelegate);
 
-    m_resultsBrowser->hideColumn(SearchSortModel::DL_LINK); // Hide url column
-    m_resultsBrowser->hideColumn(SearchSortModel::DESC_LINK);
+    m_ui->resultsBrowser->hideColumn(SearchSortModel::DL_LINK); // Hide url column
+    m_ui->resultsBrowser->hideColumn(SearchSortModel::DESC_LINK);
 
-    m_resultsBrowser->setRootIsDecorated(false);
-    m_resultsBrowser->setAllColumnsShowFocus(true);
-    m_resultsBrowser->setSortingEnabled(true);
+    m_ui->resultsBrowser->setRootIsDecorated(false);
+    m_ui->resultsBrowser->setAllColumnsShowFocus(true);
+    m_ui->resultsBrowser->setSortingEnabled(true);
 
     // Connect signals to slots (search part)
-    connect(m_resultsBrowser, SIGNAL(doubleClicked(const QModelIndex&)), this, SLOT(downloadSelectedItem(const QModelIndex&)));
+    connect(m_ui->resultsBrowser, SIGNAL(doubleClicked(const QModelIndex&)), this, SLOT(downloadSelectedItem(const QModelIndex&)));
 
     // Load last columns width for search results list
     if (!loadColWidthResultsList())
-        m_resultsBrowser->header()->resizeSection(0, 275);
+        m_ui->resultsBrowser->header()->resizeSection(0, 275);
 
     // Sort by Seeds
-    m_resultsBrowser->sortByColumn(SearchSortModel::SEEDS, Qt::DescendingOrder);
+    m_ui->resultsBrowser->sortByColumn(SearchSortModel::SEEDS, Qt::DescendingOrder);
+
+    fillFilterComboBoxes();
+
+    updateFilter();
+
+    connect(m_ui->filterMode, SIGNAL(activated(int)), this, SLOT(updateFilter()));
```

Comment:
I think it's a good practice to name the UI elements with the Class name suffix. For example: `filterModeCombo`.